lib/utils/job-utils/racadm-parser.js

```diff
@@ -0,0 +1,229 @@
+// Copyright 2016, EMC, Inc.
+
+"use strict";
+
+var di = require('di');
+
+module.exports = parseRacadmDataFactory;
+di.annotate(parseRacadmDataFactory, new di.Provide('JobUtils.RacadmCommandParser'));
+di.annotate(parseRacadmDataFactory,
+    new di.Inject(
+        'Assert',
+        '_'
+    )
+);
+
+function parseRacadmDataFactory(
+    assert,
+    _
+) {
+
+    function RacadmCommandParser() {}
+
+    /**
+     * Parser software inventory list to get standard software list json format.
+     *
+     * @param {string} softwareListData
+     */
+    RacadmCommandParser.prototype.getSoftwareList = function(softwareListData) {
+        var softwareInventory = {};
+
+        var lines = softwareListData.trim().split('\n');
+        var filteredLines = _.filter(lines, function(line){
+            return line.indexOf('=') !== -1 && line.indexOf('ComponentType') === -1;
+        });
+
+        if (filteredLines.length % 4 !== 0){
+            throw "software list data is not aligned in correct way";
+        }
+        var groupedData = _.chunk(filteredLines, 4);
+
+        //Example of groupedData Element
+        //data[0] - 'ElementName = Intel(R) Ethernet 10G X520 LOM - 00:8C:FA:F3:78:30'
+        //data[1] - 'FQDD = NIC.Embedded.1-1-1'
+        //data[2] - 'InstallationDate = 2015-11-26T06:54:17Z'
+        //data[3] - 'Current Version = 16.5.0'
+        _.forEach(groupedData, function(data){
+            var row = [];
+            //Example of row
+            //row[0] - ['elementName', 'Intel(R) Ethernet 10G X520 LOM - 00:8C:FA:F3:78:30']
+            //row[1] - ['fQDD', 'NIC.Embedded.1-1-1']
+            //row[2] - ['installationDate', '2015-11-26T06:54:17Z']
+            //row[3] - ['currentVersion', '16.5.0']
+            _.forEach(data, function(line){
+                var splitedLine = line.split('='),
+                    name = splitedLine[0].trim().replace(' ', '');
+                var lowerFirstChar = name.substr(0,1).toLowerCase().concat(name.substr(1));
+                row.push([lowerFirstChar, splitedLine[1].trim()]);
+            });
+            var deviceName = row[1][1].split(':')[0];
+            var deviceInfo = {
+                elementName: '',
+                FQDD: '',
+                installationDate: '',
+                currentVersion: '',
+                rollbackVersion: '',
+                availableVersion: ''
+            };
+            if (softwareInventory.hasOwnProperty(deviceName)){
+                deviceInfo = softwareInventory[deviceName];
+                if (deviceInfo.installationDate === 'NA'){
+                    deviceInfo.installationDate = row[2][1];
+                }
+                if (!deviceInfo[row[3][0]]){
+                    deviceInfo[row[3][0]] = row[3][1];
```

Comment:
Line 73 and 74: using deviceInfo['currentVersion'] might make it easier to understand than using deviceInfo[row[3][0]])